src/ETISS.cpp

```diff
@@ -647,14 +683,19 @@ void etiss::Initializer::loadIniJIT(std::shared_ptr<etiss::CPUCore> cpu)
     cpu->set(getJIT(cfg().get<std::string>("JIT_Type", "")));
 }
 
-std::pair<std::string, std::string> inifileload(const std::string& s)
+std::pair<std::string, std::string> extra_parser(const std::string& s)
 {
     if (s.find("-i") == 0)
     {
         std::string inifile;
         inifile = s.substr(2);
         etiss_loadIni(inifile);
-        return make_pair(std::string(), std::string());
+    }
+    if (s.find("-p") == 0)
+    {
+        std::string pluginName;
+        pluginName = s.substr(2); //Create a global list with all plugin names. Then iterate over it and load it after command line and ini file is completely parsed by sending options along with the plugin name.
+        pluginList.push_back(pluginName);
```

Comment:
https://www.boost.org/doc/libs/1_75_0/doc/html/program_options/howto.html

if i understand that first section correctly, you could instead do `return make_pair("pluginToLoad", pluginName);` and that should automatically be put into the `vm`. or would that just keep overwriting the same value?

this way you could eliminate the global variable `pluginList`.

Reply:
Yes that would overwrite it. I however changed this part and used multitokens in program options so that all plugins can be given at once as a vector on the command line. Now we do not need a separate global list. 

For example the Logger and gdbserver plugins can be loaded together from the command line as `-p Logger gdbserver`. This vector is stored in vm and can be directly accessed from there in any function. 

I hope this resolves the issue.

Reply:
this looks like a very good solution, thanks!

Reply:
Great! I've made all other changes. Please take a look and let me know if any other change is required.